core/src/main/java/com/linecorp/armeria/server/docs/ServiceInfo.java

```diff
@@ -96,10 +122,39 @@ public String name() {
      * Returns the metadata about the methods available in the service.
      */
     @JsonProperty
-    public Set<MethodInfo> methods() {
+    public List<MethodInfo> methods() {
         return methods;
     }
 
+    /**
+     * Duplicate {@link MethodInfo} with matching (name, httpMethod) are merged.
+     * Merge targets are endpoints since {@code exampleHttpHeaders}, {@code exampleRequests},
+     * and {@code docString} cannot be different for a single method.
```

Comment:
A little bit of rephrasing:

```
Merges the {@link MethodInfo}s with the same method name and {@link HttpMethod} pair
into a single {@link MethodInfo}. Note that only the {@link EndpointInfo}s are merged
because the {@link MethodsInfo}s being merged always have the same
{@code exampleHttpHeaders} and {@code exampleRequests}.
```